vizro-core/docs/pages/user-guides/custom-figures.md

```diff
@@ -0,0 +1,108 @@
+# How to create custom figures
+
+This guide explains how to create custom figures, which is useful when you need a figure that reacts to controls and actions.
+
+There are two use cases where you might want to use custom figures:
+
+* You need a figure that isn't available in our pre-defined figure functions [`vizro.figures`](../API-reference/figure-callables.md).
+* You need a figure that doesn't fit into the existing predefined components ([`Graph`][vizro.models.Graph], [`Table`][vizro.models.Table] or [`AgGrid`][vizro.models.AgGrid]).
+
+Use the following approach, which is similar to creation of a [custom chart](../user-guides/custom-charts.md), do the following:
+
+1. Define a function that returns a `Dash` component.
+2. Decorate it with the `@capture("figure")` respectively.
+3. The function must accept a `data_frame` argument (of type `pandas.DataFrame`).
+4. The figure should be derived from and require only one `pandas.DataFrame`. Additional dataframes from other arguments will not react to dashboard components such as `Filter`.
+
+<!-- vale off -->
+The example below shows how to create a custom figure — a list of cards created from a `pandas.DataFrame` where the
+number of cards displayed dynamically adjusts based on a `vm.Parameter`.
+
+!!! example "Dynamic number of cards"
+    === "app.py"
+        ```py
+        from typing import List, Optional
+
+        import pandas as pd
+        import vizro.models as vm
+        from vizro import Vizro
+        from vizro.models.types import capture
+
+        df = pd.DataFrame(
+            {
+                "text": [
+                    "Lorem ipsum dolor sit amet, consetetur sadipscing no sea elitr sed diam nonumy.",
+                    "Sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat.",
+                    "Sed diam voluptua. At vero eos et accusam et justo no duo dolores et ea rebum.",
+                    "Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.",
+                    "Lorem ipsum dolor sit amet, consetetur sadipscing no sea est elitr dolor sit amet.",
+                    "Sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat.",
+                ]
+                * 2
+            }
+        )
+
+
+        @capture("figure")  # (1)!
+        def multiple_cards(data_frame: pd.DataFrame, n_rows: Optional[int] = 1) -> List[vm.Card]:  # (2)!
+            """Creates a list with a variable number of `vm.Card` components from the provided data_frame.
+
+            Args:
+                data_frame: Data frame containing the data.
+                n_rows: Number of rows to use from the data_frame. Defaults to 1.
+
+            Returns:
+                List of dbc.Card objects generated from the data.
+
+            """
+            texts = data_frame.head(n_rows)["text"]
+            return [vm.Card(text=f"### Card #{i+1}\n{text}").build() for i, text in enumerate(texts)]
```

Comment:
This doesn't match the type hint! `vm.Card.build` returns `dbc.Card`.

When I was writing this example code I thought about two ways of doing it:
1. do `vm.Card.build`: means you don't need to understand any more Dash, just Vizro, but it uses the `build` method which maybe looks weird and means creating many new models on the fly
2. do `dbc.Card` directly: means you need to understand Dash but doesn't add the complications of Vizro models

In the case of answering the user I chose to go for 1, but here I think we should probably go for option 2. Otherwise it's not clear how you can use a custom figure to inject an arbitrary dash component - it just looks like you are still restricted to vizro models.

We should also consider whether we want to encourage people to return a list of Dash components or whether they should actually be forced to return one. Probably we should just restrict to only one like we do everywhere else (even though a list would probably still work fine because it just gets put in a html.Div or dcc.Loading or whatever afterewards).

So I'd suggest something like this:
```suggestion
            return html.Div(dbc.Card(dcc.Markdown(f"### Card #{i+1}\n{text}")) for i, text in enumerate(texts))
            # or if we think allowing list is ok
            return [dbc.Card(dcc.Markdown(f"### Card #{i+1}\n{text}")) for i, text in enumerate(texts)]
```
and change the type hint accordingly.

Reply:
Ah, yes - that makes sense!